src/main/java/io/vertx/ext/couchdb/impl/CouchdbClientImpl.java

```diff
@@ -81,26 +74,57 @@ public Future<Buffer> rawCall(JsonObject params) {
     initClient();
     HttpRequest<Buffer> request = client.request(method, path);
 
-    if (this.credentials != null) {
-      request.authentication(this.credentials);
+    if (clientOptions.getCredentials() != null) {
+      request.authentication(clientOptions.getCredentials().toAuthCredentials());
     }
     request.send()
```

Comment:
need at some time account for streaming bodies here

Reply:
sure, but it might not be included in this PR yet